src/core.ts

```diff
@@ -70,15 +74,32 @@ export type Observable<T> = {
 export type FromInput<T> = Promise<T> | Stream<T> | Array<T> | Observable<T>;
 
 // mutates the input
-function internalizeProducer<T>(producer: Producer<T>) {
+function internalizeProducer<T>(producer: Producer<T> | ProducerFn<T>) {
   (producer as InternalProducer<T> & Producer<T>)._start =
     function _start(il: InternalListener<T>) {
       (il as InternalListener<T> & Listener<T>).next = il._n;
       (il as InternalListener<T> & Listener<T>).error = il._e;
-      (il as InternalListener<T> & Listener<T>).complete = il._c;
-      this.start(il as InternalListener<T> & Listener<T>);
+      if (typeof producer === 'function') {
+        let _stoped = false;
+        (il as InternalListener<T> & Listener<T>).complete = function _stop () {
```

Comment:
Performance problem: since the function `_stop` is created inline here, every time this producer starts, a new object allocation happens.